wfl/Database/MongoDBManager.cs

```diff
@@ -19,62 +20,66 @@ public MongoDBManager()
             db = client.GetDatabase("WFL");
         }
 
-        public async Task<Restaurant> InsertRestaurantAsync(RestaurantDto r)
+        private string GenerateCollectionName<T>() where T: new()
         {
-            IMongoCollection<Restaurant> collection = db.GetCollection<Restaurant>("restaurants");
-            Restaurant restaurant = new Restaurant()
+            T t = new T();
+            string type = t.GetType().ToString();
```

Comment:
You can also use `t.GetType().Name` here. I beleive its `.Name`, but there is a property on the `Type` object that gets you the name of the class. Following this pattern, you are going to run into issues if the entity is a generic type because the name will be something like `SomeClass<,,>` or something like that.